tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Web/wwwroot/app/directives/common.directive.js

```diff
@@ -512,6 +512,153 @@
+                                for (var i = 0; i < scope.input.values.length; i++) {
+                                    radioButtontText += '<input name="radioGroup' + scope.input.values[i].fieldInternalName + '" type="radio"  ng-model="input.value" value="' + scope.input.values[i].choiceValue + '"><label id="">' + scope.input.values[i].choiceValue + ' </label>'
+                                }
+                            }
+                            el.append('<div style="padding-right:190px;float:right;">' + radioButtontText + '</div>')
+                            break;
+                        case 'dropdown':
+                            el.append('<div style="padding-right:190px;float:right;"><select id="' + scope.input.fieldInternalName + '" class="directiveFormFields" ng-class="{errorBorder: (input.value == undefined && cm.addFieldReq == true)}" required ng-model="input.value" ng-options=" x.choiceValue   for x in  input.values "> <option value="" label="- Select -"></option></select></div>')
+                            break;
+                        case 'datetime':
+                            el.append('<div style="padding-right:190px;float:right;"> <input id="' + scope.input.fieldInternalName + '" class="directiveFormFields" ng-class="{errorBorder: (input.value == undefined && cm.addFieldReq == true)}" required type="text" class="calendar form-control " uib-datepicker-popup="MM/dd/yyyy" data-ng-model="input.value"  is-open="opened" placeholder="mm/dd/yyyy"  data-ng-model="" datepicker-options="dateOptions" ng-required="true" close-text="Close" readonly  ng-click="open1()"  /> </div>')
+                            break;
+                        case 'multichoice':
+                            el.append('<div style="padding-right:190px;float:right;"><select  id="' + scope.input.fieldInternalName + '" class="multiSelectHeight directiveFormFields" required style="height:100px;" multiple ng-model="input.value" ng-class="{errorBorder: (input.value == undefined && cm.addFieldReq == true)}" ng-options="x.choiceValue for x  in input.values "> <option value="" label="- Select -"></option></select></div>')
+                            break;
+                    }
+
+                    $compile(el)(scope);
+                    element.append(el);
+                }
+            },
+            controller: function ($scope) {
+                $scope.open1 = function ($event) {
+                    if ($event) {
+                        $event.preventDefault();
+                        $event.stopPropagation();
+                    }
+                    $scope.opened = $scope.opened ? false : true;
+                };
+                $scope.dateOptions = {
+
+                    formatYear: 'yy',
+                    startingDay: 1
+                };
+                $scope.opened = false;
+            }
+        }
+    }
+
+    'use strict';
+    function extramatterpropertiefiledsinsettings($compile) {

```

Comment:
Purpose: new directive to render the HTML for the extra fields on Settings page.